src/lib/netlist/plib/ptimed_queue.h

```diff
@@ -179,26 +120,132 @@ namespace plib {
 
 		// save state support & mame disassembler
 
-		const T *list_pointer() const noexcept { return &m_list[1]; }
-		std::size_t size() const noexcept { return narrow_cast<std::size_t>(m_end - &m_list[1]); }
-		const T & operator[](std::size_t index) const noexcept { return m_list[ 1 + index]; }
-	private:
-		using mutex_type       = pspin_mutex<TS>;
-		using lock_guard_type  = std::lock_guard<mutex_type>;
+		constexpr const T *list_pointer() const noexcept { return &m_list[1]; }
+		constexpr std::size_t size() const noexcept { return narrow_cast<std::size_t>(m_end - &m_list[1]); }
+		constexpr const T & operator[](std::size_t index) const noexcept { return m_list[ 1 + index]; }
 
-		mutex_type               m_lock;
+	private:
+		PALIGNAS(PALIGN_CACHELINE)
 		T *                      m_end;
-		plib::arena_vector<A, T> m_list;
+		plib::arena_vector<A, T, PALIGN_CACHELINE> m_list;
 
 	public:
 		// profiling
 		// FIXME: Make those private
-		pperfcount_t<true> m_prof_sortmove; // NOLINT
+		pperfcount_t<true> m_prof_sort_move; // NOLINT
 		pperfcount_t<true> m_prof_call; // NOLINT
 		pperfcount_t<true> m_prof_remove; // NOLINT
 	};
 
-	template <class A, class T, bool TS>
+	template <class A, class T>
+	template<bool KEEPSTAT, typename... Args>
+	inline constexpr void timed_queue_linear<A, T>::emplace (Args&&... args) noexcept
+	{
+		T * i(m_end);
+		*i = T(std::forward<Args>(args)...);
+		//if (i->object() != nullptr && exists(i->object()))
+		//  printf("Object exists %s\n", i->object()->name().c_str());
+		m_end++;
+		if constexpr (!KEEPSTAT)
+		{
+			for (; *(i-1) < *i; --i)
+			{
+				std::swap(*(i-1), *(i));
+			}
+		}
+		else
+		{
+			for (; *(i-1) < *i; --i)
+			{
+				std::swap(*(i-1), *(i));
+				m_prof_sort_move.inc();
+			}
+			m_prof_call.inc();
+		}
+	}
+
+	template <class A, class T>
+	template<bool KEEPSTAT>
+	inline constexpr void timed_queue_linear<A, T>::push(T && e) noexcept
+	{
+#if 0
```

Comment:
Please comment on the purpose of the disabled block of code – it can be confusing to encounter a `#if 0` without any context on why the two variants are present, and why the one that’s disabled has been disabled.